ios/RiveReactNativeView.swift

```diff
@@ -211,12 +224,47 @@ class RiveReactNativeView: RCTView, RivePlayerDelegate, RiveStateMachineDelegate
             configureViewModelFromUrl() // TODO: calling viewModel?.configureModel for a URL ViewModel throws. Requires further investigation. Currently recreating the whole ViewModel for certain prop changes.
         }
     }
+    
+    private func updateReferencedAssets(incomingReferencedAssets: NSDictionary?) {
+        if let referencedAssets = incomingReferencedAssets?.copy() as? NSDictionary, let cachedReferencedAssets = previousReferencedAssets?.copy() as? NSDictionary {
+            let referencedKeys = Set(referencedAssets.allKeys as! [String])
+            let cachedKeys = Set(cachedReferencedAssets.allKeys as! [String])
+            // The keys are different, reloading the whole file
+            if referencedKeys != cachedKeys {
+                requiresLocalResourceReconfigure = true
+            } else {
+                var hasChanged = false
+                for (key, value) in referencedAssets {
+                    if let keyString = key as? String,
+                       let cachedValue = cachedReferencedAssets[keyString] as? NSDictionary,
+                       let newValue = value as? NSDictionary,
+                       !cachedValue.isEqual(newValue) {
+                        hasChanged = true
+                        if let source = newValue["source"] as? NSDictionary,
+                           let asset = cachedFileAssets[keyString],
+                           let factory = cachedRiveFactory {
+                            loadAsset(source: source, asset: asset, factory: factory)
+                        }
+                    }
+                }
+                if hasChanged && viewModel?.isPlaying == false {
+                    viewModel?.play() // manually calling play to force an update, ideally want to do a single advance
```

Comment:
@dskuza need your input here especially. Ideally, I just want to advance and draw by a single frame if the State Machine is already settled (or it's paused).

What is interesting though is calling `play` in this scenario restarts everything from scratch. To me that should only ever happen if you called `stop` then `play`?